### PR TITLE
[FEATRURE] Add fusing FullyConnected with element-wise add

### PR DESCRIPTION
**Pull request by anko-intel**
## Description ##
Implement fuzing FullyConnected with element-wise add for floating point grapth (MKLDNN backend) and for quantization_mode='full' where all calculation are made on int8 (MKLDNN_QUANTIZE backend)

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [ ] Code is well-documented

## Comments ##
It improves  performance of FC+add up to 25% for [microbenchmark](https://gist.github.com/anko-intel/4c285456c97807dd49ab76112b905296) run on AWSm6i.32xlarge:

MKLDNN:   float32 :

shape | hidden layers | mean time [ms] | mean tim [ms] | improvement [%]
-- | -- | -- | -- | --
  |   | before | after |  
(     1,  224) | 512 | 0.124 | 0.113 | 9
(     1,  224) | 1024 | 0.125 | 0.118 | 6
(     1,  224) | 4096 | 0.126 | 0.121 | 4
(    16, 1024) | 512 | 0.163 | 0.147 | 10
(    16, 1024) | 1024 | 0.174 | 0.165 | 5
(    16, 1024) | 4096 | 0.3 | 0.268 | 11
(    32, 4096) | 512 | 0.266 | 0.246 | 8
(    32, 4096) | 1024 | 0.386 | 0.363 | 6
(    32, 4096) | 4096 | 1.381 | 1.333 | 3
( 512,    512) | 512 | 0.321 | 0.263 | 18
( 512,    512) | 1024 | 0.462 | 0.354 | 23
( 512,    512) | 4096 | 1.161 | 0.871 | 25

MKLDNN_QUANTIZE: quantize_mode='smart' :

shape | hidden layers | mean time [ms] | mean tim [ms] | improvement [%]
-- | -- | -- | -- | --
  |   | before | after |  
(     1,  224) | 512 | 0.224 | 0.223 | 0
(     1,  224) | 1024 | 0.228 | 0.228 | 0
(     1,  224) | 4096 | 0.222 | 0.216 | 3
(    16, 1024) | 512 | 0.281 | 0.276 | 2
(    16, 1024) | 1024 | 0.292 | 0.301 | -3
(    16, 1024) | 4096 | 0.319 | 0.313 | 2
(    32, 4096) | 512 | 0.348 | 0.345 | 1
(    32, 4096) | 1024 | 0.382 | 0.375 | 2
(    32, 4096) | 4096 | 0.503 | 0.502 | 0
( 512,    512) | 512 | 0.362 | 0.364 | -1
( 512,    512) | 1024 | 0.451 | 0.451 | 0
( 512,    512) | 4096 | 0.83 | 0.816 | 2

MKLDNN_QUANTIZE: quantize_mode='full':

shape | hidden layers | mean time [ms] | mean tim [ms] | improvement [%]
-- | -- | -- | -- | --
  |   | before | after |  
(     1,  224) | 512 | 0.24 | 0.217 | 10
(     1,  224) | 1024 | 0.234 | 0.217 | 7
(     1,  224) | 4096 | 0.281 | 0.236 | 16
(    16, 1024) | 512 | 0.343 | 0.284 | 17
(    16, 1024) | 1024 | 0.348 | 0.299 | 14
(    16, 1024) | 4096 | 0.396 | 0.33 | 17
(    32, 4096) | 512 | 0.414 | 0.354 | 14
(    32, 4096) | 1024 | 0.45 | 0.389 | 14
(    32, 4096) | 4096 | 0.598 | 0.516 | 14
( 512,    512) | 512 | 0.44 | 0.365 | 17
( 512,    512) | 1024 | 0.541 | 0.472 | 13
( 512,    512) | 4096 | 0.935 | 0.825 | 12

For quantize_mode='smart' results are within 3% measurement noise as for this  mode FC is already fused with element-wise add by #20400

### CODE REVIEW

**Comment by anko-intel** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by anko-intel** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by anko-intel** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by anko-intel** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by anko-intel** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by anko-intel** [comment]
@mxnet-bot run ci [unix-gpu]

File: src/operator/subgraph/mkldnn/mkldnn_fc_sum_fuse.h [hunk 1]
```diff
@@ -163,13 +186,21 @@ class SgMKLDNNFCSumFuseProperty : public SubgraphProperty {
...
       auto fc_orginal = fc_node->attrs.subgraphs[0]->outputs[0].node;
+
       if (fc_orginal->op() == Op::Get("FullyConnected")) {
         nnvm::Symbol new_sym;
-        nnvm::NodeEntry& ew_input_with_fc = (ew_add_node->inputs[1].node == fc_node)
-                                                ? ew_add_node->inputs[1]
-                                                : ew_add_node->inputs[0];
-        ew_input_with_fc.node             = fc_orginal;
-        new_sym.outputs.emplace_back(ew_add_node);
+        nnvm::ObjectPtr n = nnvm::Node::Create();
```
Comment (bgawrych) [inline src/operator/subgraph/mkldnn/mkldnn_fc_sum_fuse.h:192]: Consider adding comment why elemwise_add node must be placed in subgraph (InferShape/InferType)

**Review by bgawrych — APPROVED** [review]

File: src/operator/subgraph/mkldnn/mkldnn_fc_sum_fuse.h (hunk 1 above)
Comment (anko-intel) [reply src/operator/subgraph/mkldnn/mkldnn_fc_sum_fuse.h:194]: Done

**Review by akarbown — APPROVED** [review]

**Pull request merged**